staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -196,11 +196,15 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 	if len(requestInfo.Name) == 0 && requestInfo.Verb == "get" {
 		// Assumes v1.ListOptions
 		// Duplicates logic of Convert_Slice_string_To_bool
-		switch strings.ToLower(req.URL.Query().Get("watch")) {
-		case "false", "0", "":
+		if values := req.URL.Query()["watch"]; len(values) > 0 {
+			switch strings.ToLower(values[0]) {
+			case "false", "0":
+				requestInfo.Verb = "list"
+			default:
+				requestInfo.Verb = "watch"
+			}
+		} else {
```

Comment:
Ah, since we have to load the values, we'd have to have two nested switches (we need to conditionalize the ToLower on values being non empty)